File: sol/solution/0101-0200/0102/Solution.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
class Solution {
  List<List<int>> res = [];
  List<List<int>> levelOrder(TreeNode? root) {
    dfs(root, 0);
    return res;
  }
  dfs(TreeNode? node, int level){
    if (node == null) return;
    if (res.length <= level) res.add([]);
    res[level].add(node.val);
    dfs(node.left, level+1);
    dfs(node.right, level+1);
  }
}

File: sol/solution/0101-0200/0102/Solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        var res = new List<IList<int>>();
        if (root == null) return res;
        var q = new Queue<TreeNode>();
        q.Enqueue(root);
        while (q.Count > 0) {
            int sz = q.Count;
            var level = new List<int>();
            for (int i = 0; i < sz; i++) {
                var node = q.Dequeue();
                level.Add(node.val);
                if (node.left != null) q.Enqueue(node.left);
                if (node.right != null) q.Enqueue(node.right);
            }
            res.Add(level);
        }
        return res;
    }
}
